test/cpp/jit/test_gpu.cpp

```diff
@@ -2155,7 +2172,7 @@ void testGPU_FusionRFactorReplay() {
 
   // Do math with it, it returns a `Val*` but can be static_casted back to
   // TensorView
-  TensorView* tv1 = sum(tv0, {1});
+  TensorView* tv1 = static_cast<TensorView*>(sum(tv0, {1}));
```

Comment:
nitpick: we should use `as` instead of `static_cast` for consistency.
Moreover, reduction only works on tensor, maybe it should return `TensorView*` directly?

Reply:
`sum` returns `TensorView*` so no need to cast anymore.

Reply:
Fixed all static_casts in test_gpu.